big_tests/tests/sasl_external_SUITE.erl

```diff
@@ -85,6 +122,47 @@ cert_with_cn_no_xmpp_addrs_request_name_empty(C) ->
 
     escalus_connection:stop(Client).
 
+self_signed_cert_fails_to_authenticate_with_tls(C) ->
+    self_signed_cert_fails_to_authenticate(C, escalus_tcp).
+
+self_signed_cert_fails_to_authenticate_with_ws(C) ->
+    self_signed_cert_fails_to_authenticate(C, escalus_ws).
+
+self_signed_cert_fails_to_authenticate_with_bosh(C) ->
+    self_signed_cert_fails_to_authenticate(C, escalus_bosh).
+
+self_signed_cert_fails_to_authenticate(C, EscalusTransport) ->
+    F = fun() ->
+		UserSpec = generate_user(C, "alice-self-signed", EscalusTransport),
+		{ok, Client, _} = escalus_connection:start(UserSpec),
+		escalus_connection:stop(Client)
+	end,
+    %% We spawn the process trying to connect because otherwise the testcase may crash
+    %% due linked process crash (client's process are started with start_link)
+    Pid = spawn(F),
+    MRef = erlang:monitor(process, Pid),
+
+    receive
+	{'DOWN', MRef, process, Pid, Reason} ->
```

Comment:
Won't this test pass even if the connection succeeds? `escalus_connectoin:start(UserSpec)` will return after connecting, then we close the client by `escalus_connection:stop(Client)` and then the process finishes with `ok`, so we'll get a `'DOWN'` message all the same.

Reply:
Good catch! I'll fix that.